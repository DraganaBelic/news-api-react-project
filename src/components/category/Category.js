import React, { useEffect, Fragment } from "react";
import { useSelector, useDispatch } from "react-redux";
import { TopNewsWrapperStyle } from "../topNews/TopNewsStyle";
import Thumbnail from "../thumbnail/Thumbnail";
import { useTranslation } from "react-i18next";
import { fetchAllByCategory } from "../../api/apiUtil";
import LoaderComponent from "../../general/LoaderComponent";

const Category = () => {
  const currentCountry = useSelector((state) => state.country.currentCountry);
  const fetchData = useSelector((state) => state.category);
  const response = fetchData.response;
  const category = useSelector((state) => state.category.currentCategory);
  const dispatch = useDispatch();
  const { t } = useTranslation();
  
  useEffect(() => {
    dispatch(fetchAllByCategory(currentCountry, category));
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [currentCountry, category]);
  let fetchedNews =
    response && response.articles
      ? response.articles.map((tn, i) => {
          return <Thumbnail key={i} article={tn} />;
        })
      : "";
  return (
    <Fragment>
      <h2>
        {t("top")} {t(category)} {t("newsFrom")} {t(currentCountry)}
      </h2>
      {fetchData && fetchData.loading ? (
        <LoaderComponent />
      ) : (
        <TopNewsWrapperStyle>{fetchedNews}</TopNewsWrapperStyle>
      )}
    </Fragment>
  );
};

export default Category;
