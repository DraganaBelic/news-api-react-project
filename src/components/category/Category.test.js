import React from "react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import thunk from "redux-thunk";
import { MemoryRouter } from "react-router-dom";
import Category from "./Category";

const middlewares = [thunk];
const initialState = {
  category: {
    currentCategory: "science",
    response: {
      articles: [
        {
          title: "Article title 1",
          urlToImage: "urlToImage",
          description: "Article description 1",
        },
        {
          title: "Article title 2",
          urlToImage: "urlToImage",
          description: "Article description 2",
        }
      ],
    },
    loading: true
  },
  country: {
    currentCountry: "gb",
  },
};

export const mockedStore = (state = initialState) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state);
};

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (s) => s }),
}));
describe("Test Category component", () => {
  const store = mockedStore();
  const { container } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Category />
      </MemoryRouter>
    </Provider>
  );
  it("should render with given state from Redux store", () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});
