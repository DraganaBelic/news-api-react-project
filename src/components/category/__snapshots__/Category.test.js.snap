// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Category component should render with given state from Redux store 1`] = `
<h2>
  top
   
  science
   
  newsFrom
   
  gb
</h2>
`;
